src/Chainweb/Mempool/Mempool.hs

```diff
@@ -428,62 +522,51 @@ syncMempools' log0 us localMempool remoteMempool = sync
     deb :: Text -> IO ()
     deb = log0 Debug
 
-    sync = finally (initialSync >>= subsequentSync) (deb "sync exiting")
+    sync = finally initialSync (deb "sync exiting")
 
     initialSync = do
-        deb "Get full list of pending hashes from remote"
+        mHistory <- fetchGlobalSyncHistory peer
```

Comment:
Given that you fetch peer sync history between the current mempool and a peer, is it possible for the actual highwater mark of the mempool to be different than `ourHwMark` that you get from the past? Is this a problem?

Reply:
No, because it's stored in RAM and the server nonce is changed at restart. Even if server nonces were different, the default behaviour is "return everything" on nonce mismatch.